File: frontend/src/microservices/flight/objects/FlightDailyLowestPricePlannerResponse.ts
// 本文件定义 flight 模块的 `FlightDailyLowestPricePlannerResponse`，作为单日价格响应数据并提供 JSON 编解码。

export type FlightDailyLowestPricePlannerResponse = {
  date: string
  lowestPrice: string | null
  currency: string | null
}

export const flightDailyLowestPricePlannerResponseFromJson = (json: string): FlightDailyLowestPricePlannerResponse =>
  JSON.parse(json) as FlightDailyLowestPricePlannerResponse

export const flightDailyLowestPricePlannerResponseToJson = (value: FlightDailyLowestPricePlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightDailyLowestPricePlannerResponse.scala
// FlightDailyLowestPricePlannerResponse defines one daily price entry for the flight planner.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightDailyLowestPricePlannerResponse(
    date: String,
    lowestPrice: Option[String],
    currency: Option[String]
)

object FlightDailyLowestPricePlannerResponse:
  given sourceEncoder: Encoder[FlightDailyLowestPricePlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FlightDailyLowestPricePlannerResponse] = deriveDecoder

